cmd/legacy_param_converter.go

```diff
@@ -0,0 +1,155 @@
+			"custom-endpoint":               flags.CustomEndpoint,
+			"experimental-enable-json-read": flags.ExperimentalEnableJsonRead,
+			"http-client-timeout":           flags.HttpClientTimeout,
+			"limit-bytes-per-sec":           flags.EgressBandwidthLimitBytesPerSecond,
+			"limit-ops-per-sec":             flags.OpRateLimitHz,
+			"max-conns-per-host":            flags.MaxConnsPerHost,
+			"max-idle-conns-per-host":       flags.MaxIdleConnsPerHost,
+			"sequential-read-size-mb":       flags.SequentialReadSizeMb,
+		},
+		"gcs-retries": map[string]interface{}{
+			"max-retry-sleep": flags.MaxRetrySleep,
+			"multiplier":      flags.RetryMultiplier,
+		},
+		"implicit-dirs": flags.ImplicitDirs,
+		"list": map[string]interface{}{
+			"kernel-list-cache-ttl-secs": flags.KernelListCacheTtlSeconds,
+		},
+		"logging": map[string]interface{}{
+			"file-path": flags.LogFile,
+			"format":    flags.LogFormat,
+		},
+		"metadata-cache": map[string]interface{}{
+			"deprecated-stat-cache-capacity":          flags.StatCacheCapacity,
+			"deprecated-stat-cache-ttl":               flags.StatCacheTTL,
+			"deprecated-type-cache-ttl":               flags.TypeCacheTTL,
+			"enable-nonexistent-type-cache":           flags.EnableNonexistentTypeCache,
+			"experimental-metadata-prefetch-on-mount": flags.ExperimentalMetadataPrefetchOnMount,
+		},
+		"metrics": map[string]interface{}{
+			"stackdriver-export-interval": flags.StackdriverExportInterval,
+		},
+		"monitoring": map[string]interface{}{
+			"experimental-opentelemetry-collector-address": flags.OtelCollectorAddress,
+		},
+		"only-dir": flags.OnlyDir,
+	}
+
+	// Use decoder to convert flagStorage to cfg.Config.
+	decoderConfig := &mapstructure.DecoderConfig{
+		DecodeHook: cfg.DecodeHook(),
+		Result:     resolvedConfig,
+		TagName:    "yaml",
+	}
+	decoder, err := mapstructure.NewDecoder(decoderConfig)
+	if err != nil {
+		return nil, fmt.Errorf("mapstructure.NewDecoder: %v", err)
+	}
+	// Decoding flags.
+	err = decoder.Decode(structuredFlags)
+	if err != nil {
+		return nil, fmt.Errorf("decoder.Decode(structuredFlags): %v", err)
+	}
+
+	// If config file is not present, no need to decode it.
+	if legacyConfig == nil || reflect.DeepEqual(*legacyConfig, config.MountConfig{}) {
+		return resolvedConfig, nil
+	}
+
+	// Save overlapping flags in a map to override the config value later.
+	overlapFlags := map[string]interface{}{
+		"log-file":                   resolvedConfig.Logging.FilePath,
+		"log-format":                 resolvedConfig.Logging.Format,
+		"ignore-interrupts":          resolvedConfig.FileSystem.IgnoreInterrupts,
+		"anonymous-access":           resolvedConfig.GcsAuth.AnonymousAccess,
+		"kernel-list-cache-ttl-secs": resolvedConfig.List.KernelListCacheTtlSecs,
+	}
+
+	// Decoding config to the same config structure.
+	err = decoder.Decode(legacyConfig)
+	if err != nil {
+		return nil, fmt.Errorf("decoder.Decode(config): %v", err)
```

Comment:
It's recommended to use %w for errors - go/go-style/best-practices#error-percent-w
```suggestion
		return nil, fmt.Errorf("decoder.Decode(config): %w", err)
```